stdlib/public/runtime/Private.h

```diff
@@ -31,27 +31,22 @@ namespace swift {
 
 class TypeReferenceOwnership {
   enum : uint8_t {
-    Weak = 1 << 0,
-    Unowned = 1 << 1,
-    Unmanaged = 1 << 2,
+    Invalid,
+#define REF_STORAGE(Name, ...) \
+    Name,
+#include "swift/AST/ReferenceStorage.def"
```

Comment:
Are you sure this type isn't part of the ABI?

Reply:
I've reverted this in a forthcoming pull request update. For the record, this ABI is really shortsighted and broken. It only allows for 8 reference storage types, and the "flags" approach allows for impossible scenarios.